### PR TITLE
Fix PreviewPublicDetector

### LINKED ISSUE #189: ComposePreviewPublic not working as expected
We have recently added `compose-lint` to our lint setup and started noticing that the `ComposePreviewPublic` rule is not detecting any of our `public` previews.

As to my understanding the following compose preview should trigger a warning
```kotlin
@Preview
@Composable
fun TestComposable_Preview() {
   // some composable
}
```

For my project this is not working. I am only able to create this warning by configuring the rule as such
```xml
<lint>
    <issue id="ComposePreviewPublic" severity="error">
        <option name="preview-public-only-if-params" value="false" />
    </issue>
</lint>
```

Please find a [small sample project](https://github.com/florianPOLARSTEPS/LintCheckIssue) demonstrating the problem

### PR DESCRIPTION
**Pull request by ZacSweers**
Resolves #189

### CODE REVIEW

File: compose-lint-checks/src/test/java/slack/lint/compose/PreviewPublicDetectorTest.kt [hunk 1]
```diff
@@ -30,6 +31,7 @@ class PreviewPublicDetectorTest : BaseSlackLintTest() {
     lint().files(kotlin(code)).allowCompilationErrors().run().expectClean()
   }
 
+  @Ignore("TODO code review")
```
Comment (ZacSweers) [inline compose-lint-checks/src/test/java/slack/lint/compose/PreviewPublicDetectorTest.kt:34]: @mrmans0n @chrisbanes I was surprised to find this test that starts failing after the fix. The test seems to disagree with the check's stated purpose, do you have more details?

image

Comment (chrisbanes) [reply compose-lint-checks/src/test/java/slack/lint/compose/PreviewPublicDetectorTest.kt:34]: Not sure. I assume the second function declaration is a typo

Comment (mrmans0n) [reply compose-lint-checks/src/test/java/slack/lint/compose/PreviewPublicDetectorTest.kt:34]: Tried to remember why this was done... but I can't recall exactly. The idea I think was around previews being alright being public, private or whatever unless they had an annotated param (which made them stupid to be public then). But the meaning of the rule changed over time (adding a param to control this was added, many behavior changes happened later).

FWIW this was simplified to just check for public + preview, and ignore everything else in my repo https://github.com/mrmans0n/compose-rules/blob/main/rules/common/src/main/kotlin/io/nlopez/compose/rules/PreviewPublic.kt

You could do the same and call it a day.

Comment (ZacSweers) [reply compose-lint-checks/src/test/java/slack/lint/compose/PreviewPublicDetectorTest.kt]: [cd47d5f](https://github.com/slackhq/compose-lints/pull/206/commits/cd47d5fd50fdbadbb6302112c72f8296d7e3e275)

**Review by chrisbanes — APPROVED** [review]

**Pull request merged**